### PR TITLE
Assorted image layout / subresource checks

### PR DESCRIPTION
**Pull request by cdwfs**
The goal here was to improve validation for image layout and image resource VUs. I switched existing checks to use the appropriate VUIDs, made sure all relevant VUs were checked in all use cases (and using the same code paths), and added test cases where they were missing.

In the course of fixing CI issues, I decided to pull out the new test for `VK_KHR_shared_presentable_image` and will submit it as a separate PR after further work.

### CODE REVIEW

**Comment by mark-lunarg** [comment]
@daveh-lunarg, if you have a chance please have a look!

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -21895,19 +21895,40 @@ TEST_F(VkLayerTest, CopyImageAspectMismatch) {
 
     // Illegal combinations of aspect bits - VU 01221
```
Comment (tobine) [inline tests/layer_validation_tests.cpp]: I don't understand what "VU 01221" means here. Should probably just delete that portion of comment.

File: tests/layer_validation_tests.cpp [hunk 2]
```diff
@@ -21895,19 +21895,40 @@ TEST_F(VkLayerTest, CopyImageAspectMismatch) {
...
+    copyRegion.dstSubresource.aspectMask = VK_IMAGE_ASPECT_COLOR_BIT | VK_IMAGE_ASPECT_DEPTH_BIT;  // color must be alone
+    m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT, "VUID-VkImageSubresourceLayers-aspectMask-00167");
+    // These aspect/format mismatches are redundant but unavoidable here
     m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT, "VUID-VkImageCopy-aspectMask-00143");
+    m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT, vuid);
     vkCmdCopyImage(m_commandBuffer->handle(), color_image.handle(), VK_IMAGE_LAYOUT_GENERAL, color_image.handle(),
                    VK_IMAGE_LAYOUT_GENERAL, 1, &copyRegion);
     m_errorMonitor->VerifyFound();
+
     // Metadata aspect is illegal - VU 01222
```
Comment (tobine) [inline tests/layer_validation_tests.cpp]: Same here, delete "VU 01222".

File: layers/vk_validation_error_database.txt [hunk 3]
```diff
@@ -792,7 +792,7 @@ VALIDATION_ERROR_0a60014e~^~Y~^~CopyImageAspectMismatch~^~VkImageSubresourceLaye
...
 VALIDATION_ERROR_0a600c03~^~Y~^~Unknown~^~VkImageSubresourceLayers~^~VUID-VkImageSubresourceLayers-aspectMask-requiredbitmask~^~core~^~The spec valid usage text states 'aspectMask must not be 0' (https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#VUID-VkImageSubresourceLayers-aspectMask-requiredbitmask)~^~implicit
-VALIDATION_ERROR_0a600d48~^~N~^~None~^~VkImageSubresourceLayers~^~VUID-VkImageSubresourceLayers-layerCount-01700~^~core~^~The spec valid usage text states 'layerCount must be greater than 0' (https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#VUID-VkImageSubresourceLayers-layerCount-01700)~^~
+VALIDATION_ERROR_0a600d48~^~Y~^~None~^~VkImageSubresourceLayers~^~VUID-VkImageSubresourceLayers-layerCount-01700~^~core~^~The spec valid usage text states 'layerCount must be greater than 0' (https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#VUID-VkImageSubresourceLayers-layerCount-01700)~^~
```
[the remark this answers is not in the document]
Comment (cdwfs) [reply layers/vk_validation_error_database.txt]: Thanks, good catch; I added a couple that were already in tests, but it looks like a lot of the newly-added checks aren't tested. Fixing that now.

File: tests/layer_validation_tests.cpp (hunk 2 above)
Comment (cdwfs) [reply tests/layer_validation_tests.cpp]: I've removed all the "VU NNNNN" references in this particular test; there's a lot more elsewhere in the file, but that seems beyond the scope of this PR.

**Comment by cdwfs** [comment]
I've added tests for all the new checks that didn't previously have them (excluding the shared_presentable_image ones)

**Comment by daveh-lunarg** [comment]
LGTM.  Nice work filling in a big chunk of missing VUs.

**Comment by cdwfs** [comment]
Always a pleasure!

**Pull request merged**